Avoid name clashing in ros msg packages

Affonso-Gui opened the pull request:
Solves https://github.com/jsk-ros-pkg/jsk_roseus/issues/574.

Affonso-Gui commented:
https://github.com/jsk-ros-pkg/jsk_roseus/issues/574#issuecomment-470765507

> There are a few possible solutions for this:
> - Do not use the lisp package on ros message packages
> - Require necessary messages upon loading
> - Make sure that the symbol belongs to the designed package during operation

Adopting the first one for simplicity and to avoid re-generating the messages. Since no coding is performed from inside the ros message packages this should have no side effects (tests will confirm that for us).

Affonso-Gui commented:
Added test code

k-okada commented:
roseus does not depend on jsk_recognition_msgs, so I think it is better to create message for test code, like https://github.com/jsk-ros-pkg/jsk_roseus/pull/509/files.
I also prefer to commit test code first, and confirm the travis failed, then commit the patch (https://github.com/jsk-ros-pkg/jsk_roseus/pull/567)

Affonso-Gui commented:
@k-okada Thanks, I was insecure about the previous test code, and the link was of great help.
I added the tests to the existing `test-name*` tests since
1. Both of them test for name clashing in ros messages
1. Want to avoid having to create new test messages

Test code is also added in https://github.com/jsk-ros-pkg/jsk_roseus/pull/610

knorth55 commented:
:+1: for this PR.

Affonso-Gui commented:
@knorth55 Actually we had some test failures and possible bugs here. I'll try to have a look on this week and make any necessary changes to reactivate this
